src/pages/auth/profile/index.tsx

```diff
@@ -2,93 +2,195 @@ import React, { useEffect } from 'react';
 import Image from 'next/image';
 import Router from 'next/router';
 import { useSession } from 'next-auth/react';
-import { resetPassword } from '../../../services';
-import { Header } from '../../../components/Global';
+import toast from 'react-hot-toast';
+import { resetPassword, getUserByEmail, updateUsername } from '../../../services';
+import { Header, Input, Button } from '../../../components/Global';
+import styles from '../../../styles/pages/profile.module.scss';
 
 const ProfilePage = () => {
   const { data: session, status } = useSession();
-  const userImage = `${session?.user?.image}#site.com/image.jpg`;
+  const [hasPassword, setHasPassword] = React.useState(false);
+  const [newUsername, setNewUserName] = React.useState('');
+  const [username, setUsername] = React.useState('');
+  const [userImg, setUserImg] = React.useState('');
+  const [userEmail, setUserEmail] = React.useState('');
   const passwordMinLength = 8;
 
   const [oldPass, setOldPass] = React.useState('');
   const [password, setPassword] = React.useState('');
   const [confirmPass, setConfirmPass] = React.useState('');
 
   const handleChangePassword = async () => {
-    const email = session?.user?.email;
-
-    if (!email || !oldPass || !password || !confirmPass) return;
+    if (!userEmail || !oldPass || !password || !confirmPass) return;
     if (password !== confirmPass) return;
-
-    const success = await resetPassword({
-      email,
-      oldPassword: oldPass,
-      newPassword: password
-    });
+    try {
+      await toast.promise(
+        resetPassword({
+          email: userEmail,
+          oldPassword: oldPass,
+          newPassword: password
+        }),
+        {
+          loading: 'Changing password',
+          success: 'Password changed successfully',
+          error: err => `Error: ${err.message}`
+        }
+      );
+    } catch (err) {
+      console.error(err);
+    }
   };
 
+  useEffect(() => {
+    if (!session) return;
+
+    console.log(session)
+    setUsername(session.user.name);
+    setUserImg(session.user.image);
+    setUserEmail(session.user.email);
+  }, [session]);
+
   useEffect(() => {
     if (status === 'unauthenticated') {
       Router.push('/auth/login');
     }
   }, [status]);
 
+  useEffect(() => {
+    getUserByEmail({ email: userEmail })
+      .then((user) => {
+        if (user?.password) setHasPassword(true);
+        if (user.name) setUsername(user.name)
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }, [userEmail]);
+
+  const handleUpdateUsername = async () => {
```

Comment:
I think there's too much logic in this page, we should move utility functions to `util` files

Reply:
I don't think this is too much

Reply:
And these functions are also responsible for rendering, therefore, if we want to make this simpler, we can break them down to smaller components

Reply:
Hmm i guess now that I've taken a deeper look it's probably okay